### PR TITLE
HPCC-31428 ECL Watch v9 change ZAP dialog API endpoint

### PR DESCRIPTION
**Pull request by jeclrsg**
should use /WsWorkunits/WUCreateAndDownloadZAPInfo

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jeclrsg I wanted to make sure we tested this change before it got merged.
Take a look at this comment from the original PR: https://github.com/hpcc-systems/HPCC-Platform/pull/17476#issuecomment-1634977443

I have a feeling you're submitting form field names using dot notation and the WUcreateanddownload method might be expecting underscored notation.

**Comment by GordonSmith** [comment]
@rpastrana please re-review.

File: esp/src/src-react/components/forms/ZAPDialog.tsx [hunk 1]
```diff
@@ -192,14 +192,14 @@ export const ZAPDialog: React.FunctionComponent<ZAPDialogProps> = ({
                     if (key === "AbsoluteTimeRange") {
                         const startDate = logFilter.AbsoluteTimeRange.StartDate ? new Date(logFilter.AbsoluteTimeRange.StartDate).toISOString() : "";
                         const endDate = logFilter.AbsoluteTimeRange.EndDate ? new Date(logFilter.AbsoluteTimeRange.EndDate).toISOString() : "";
-                        formData.append("LogFilter.AbsoluteTimeRange.StartDate", startDate);
-                        formData.append("LogFilter.AbsoluteTimeRange.EndDate", endDate);
+                        formData.append("LogFilter_AbsoluteTimeRange_StartDate", startDate);
+                        formData.append("LogFilter_AbsoluteTimeRange_EndDate", endDate);
                     } else {
-                        formData.append(`LogFilter.${key}`, logFilter[key]);
+                        formData.append(`LogFilter_${key.replace(/\./, "_")}`, logFilter[key]);
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/ZAPDialog.tsx:198]: @jeclrsg should this be an url encode?

Comment (jeclrsg) [reply esp/src/src-react/components/forms/ZAPDialog.tsx]: It was unnecessary; I thought there were some fields with "." in the name, but there were not.

**Review by rpastrana — APPROVED** [review]
@jeclrsg it seems fine. 
Does it make sense to refactor the field names on the server side to match the dot notation?

**Comment by ghalliday** [comment]
@jeclrsg merging.
For future PRs please think about wording the commit message so that it describes the end effect on the user.  E.g support zap reports greater than 10MB.  The jira should also contain details of the change and how it solves the problem.
Thanks

**Pull request merged**